### ISSUE TITLE
node-red exits with status code 0 when settings.js contains an error

### ISSUE THREAD

**Issue by onjen** [node-red/node-red#4121]
### Current Behavior

```
Error loading settings file: /tmp/node_red_stage_1/settings.js
/tmp/node_red_stage_1/settings.js:61
    this is atest
         ^^

SyntaxError: Unexpected identifier
    at internalCompileFunction (node:internal/vm:74:18)
    at wrapSafe (node:internal/modules/cjs/loader:1141:20)
    at Module._compile (node:internal/modules/cjs/loader:1182:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/node-red/red.js:140:20)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
```
Still the process exits with status code 0.
```
echo $?
0
```

### Expected Behavior

The process should exit with a non-zero status code in case the settings.js file could not be loaded. This is important for example to run automated checks as part of CI/CD pipelines.

### Steps To Reproduce

- Create a syntax error in the `settings.js` file
- Start node-red
- echo the exit status `echo $?`

### Example flow

-

### Environment

- Node-RED version: 3.0.2
- Node.js version: v18.13.0
- npm version: 9.2.0
- Platform/OS: Debian testing (Linux 6.1.0-7-amd64)

Found together with @ch-sa

**Comment by Steve-Mcl** [comment]
Hi @onjen - thanks for taking the time to raise an issue.

If we are to address the exit code for this, I would suggest we address all instances of `process.exit`.  

For example, there are around 9 occurrences of `process.exit` in `red.js`. To better understand why the process exits, I propose we should look at using distinct exit codes for each "abnormal" exit.  Ideally, these would be different to common exit nodejs codes to better identify the issue occurring.

[Example](https://tldp.org/LDP/abs/html/exitcodes.html):
> According to the above table, exit codes 1 - 2, 126 - 165, and 255 [[1]](https://tldp.org/LDP/abs/html/exitcodes.html#FTN.AEN23629) have special meanings, and should therefore be avoided for user-specified exit parameters


Lets see how @knolleary and @dceejay feel about this before proceeding.

Thanks, Steve.

**Comment by Steve-Mcl** [comment]
PS: I would not necessary call this a "bug" - more of a change in behaviour therefore I would therefore propose any changes are targeted to the `dev` branch.

**Comment by onjen** [comment]
I agree that properly addressing all instances of process.exit makes more sense!

**Comment by dceejay** [comment]
I agree it makes sense to add exit codes if we can... but that we should check all 9 (?) to see if a) they can be grouped in any way and b) if any do fit within the definitions of "special" codes and re-use them if approriate.

**Comment by knolleary** [comment]
I have addressed this via https://github.com/node-red/node-red/pull/4181 as the original proposed PR isn't available any more.

There were two places where we exited with a `0` even though it was an error scenario. I have made them both exit `1` - to be consistent with the other existing error codes.

Whilst we could look at having a range of possible error codes for different scenarios, this at least addresses the immediate requirement pending any further interest or discussion in doing so.
